native/core/src/execution/jni_api.rs

```diff
@@ -485,12 +485,14 @@ pub unsafe extern "system" fn Java_org_apache_comet_Native_executePlan(
                 }
 
                 let task_ctx = exec_context.session_ctx.task_ctx();
+                // Each Comet native execution corresponds to a single Spark partition,
+                // so we should always execute partition 0.
                 let stream = exec_context
                     .root_op
                     .as_ref()
                     .unwrap()
                     .native_plan
-                    .execute(partition as usize, task_ctx)?;
+                    .execute(0, task_ctx)?;
```

Comment:
This change could impact metrics, and we lack tests in this area. I am taking a look now to understand this better and see what we can do about it.

Reply:
nm, this isn't an issue after all. ScanExec was already using partition 0 for most metrics.